pkg/components/modelserver/modelserver.go

```diff
@@ -74,21 +73,11 @@ var (
 	})
 )
 
-func NewServiceAccount() *corev1.ServiceAccount {
-	return &corev1.ServiceAccount{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: corev1.SchemeGroupVersion.String(),
-			Kind:       "ServiceAccount",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      ServiceAccountName,
-			Namespace: components.Namespace,
-			Labels:    labels,
-		},
+func NewDeployment(k *v1alpha1.Kepler) (*appsv1.Deployment, error) {
+	ms := k.Spec.ModelServer
+	if ms == nil {
+		return nil, utils.ErrNoConfig
 	}
```

Comment:
I think we shouldn't mask logical error which is that these functions should not be called when `modelserver` is nil.